fv3core/fv3core/stencils/fxadv.py

```diff
@@ -590,54 +580,51 @@ def __call__(
 
         # in the original Fortran code, uc_contra is named ut and vc_contra is vt
 
-        self._main_uc_contra_stencil(
+        self._main_uc_vc_contra_stencil(
             uc,
             vc,
             self._cosa_u,
             self._rsin_u,
-            uc_contra,
-        )
-        self._uc_contra_y_edge_stencil(uc, self._sin_sg1, self._sin_sg3, uc_contra)
-        self._main_vc_contra_stencil(
-            uc,
-            vc,
             self._cosa_v,
             self._rsin_v,
-            vc_contra,
-        )
-        self._vc_contra_y_edge_stencil(
-            vc,
-            self._cosa_v,
-            uc_contra,
-            vc_contra,
-        )
-        self._vc_contra_x_edge_stencil(vc, self._sin_sg2, self._sin_sg4, vc_contra)
-        self._uc_contra_x_edge_stencil(
-            uc,
-            self._cosa_u,
-            vc_contra,
-            uc_contra,
-        )
-        # NOTE: this is aliasing memory
-        self._uc_contra_corners_stencil(
-            self._cosa_u,
-            self._cosa_v,
-            uc,
-            vc,
-            uc_contra,
             uc_contra,
             vc_contra,
         )
-        # NOTE: this is aliasing memory
-        self._vc_contra_corners_stencil(
-            self._cosa_u,
-            self._cosa_v,
-            uc,
-            vc,
-            uc_contra,
-            vc_contra,
-            vc_contra,
-        )
+        if not self._tile_interior:
```

Comment:
May not be important here, but I was wondering if having the if statement in python vs. in stencil makes any difference?

Reply:
It should help here, we avoid a lot of python calls on top of stencil launch overheads by turning six stencil calls into one if statement.

I'd only put the if statement here because we already have separate stencils for edges and corners. If we had to separate stencils in order to use python if statements, it's much better to have joined stencils with the if statements inside.